src/main.cpp

```diff
@@ -128,20 +133,20 @@ int encode(Image &image, const std::array<std::uint8_t, 32> &password, const std
 
     // Encrypt the header
     AES aes(key, iv);
-    auto encrypted_header = new std::uint8_t[sizeof(header)];
-    aes.cbc_encrypt(&header, sizeof(header), encrypted_header);
+    auto encrypted_header = std::make_unique<uint8_t>(sizeof Header);
```

Comment:
```suggestion
    auto encrypted_header = std::make_unique<uint8_t[]>(sizeof Header);
```
Please use the array version of `std::make_unique`.